pkg/anonymization/anonymizer.go

```diff
@@ -0,0 +1,52 @@
+// Package anonymization provides Anonymizer which is used to anonymize sensitive data. At the moment,
+// anonymization is applied to all the data before storing it in the archive(see AnonymizeMemoryRecordFunction).
+// There are the following global anonymizations which can be disabled in the config:
+//   - clusterBaseDomain anonymizes cluster base domain.
+//     For example, if the cluster base domain is `openshift.example.com`,
+//     all the occurrences of this keyword will be replaced with `<CLUSTER_BASE_DOMAIN>`,
+//     `cluster-api.openshift.example.com` will become `cluster-api.<CLUSTER_BASE_DOMAIN>`
+package anonymization
+
+import (
+	"bytes"
+	"strings"
+
+	"github.com/openshift/insights-operator/pkg/config"
+	"github.com/openshift/insights-operator/pkg/config/configobserver"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// Anonymizer is used to anonymize sensitive data.
+// Config can be used to disable anonymization of particular types of data.
+type Anonymizer struct {
+	configObserver    *configobserver.Controller
+	clusterBaseDomain string
+}
+
+// NewAnonymizer creates a new instance of anonymizer with a provided config and sensitive data
+func NewAnonymizer(configObserver *configobserver.Controller, clusterBaseDomain string) *Anonymizer {
+	return &Anonymizer{
+		configObserver:    configObserver,
+		clusterBaseDomain: strings.TrimSpace(clusterBaseDomain),
+	}
+}
+
+// AnonymizeMemoryRecord takes record.MemoryRecord, removes the sensitive data from it and returns the same object
+func (anonymizer *Anonymizer) AnonymizeMemoryRecord(memoryRecord *record.MemoryRecord) *record.MemoryRecord {
+	if anonymizer.configObserver == nil {
+		return memoryRecord
+	}
+
+	conf := anonymizer.configObserver.Config().DisabledGlobalAnonymizations
+
+	if !conf.DisableClusterBaseDomainAnonymization && len(anonymizer.clusterBaseDomain) != 0 {
+		memoryRecord.Data = bytes.ReplaceAll(
+			memoryRecord.Data,
+			[]byte(anonymizer.clusterBaseDomain),
+			[]byte(config.ClusterBaseDomainPlaceholder),
+		)
```

Comment:
I really like the idea of doing this in one place. However I can see one limitation right now. The result of this is e.g:
`https://api-int.<CLUSTER_BASE_DOMAIN>:6443/api/v1/namespaces/openshift-kube-scheduler/configmaps/kube-scheduler?timeout=10s` 
I guess we should obfuscate the whole string, but again let's discuss and double check.

Reply:
Well, I don't see any sensitive information in the namespace and configmap name.

Reply:
Btw, it's not a limitation of this approach. If we do need to anonymize more that just a base domain(the whole domain or the whole URL) we can use a regular expression to find a potential match and then check if it contains cluster base domain and anonymize the whole thing.